helix-view/src/document.rs

```diff
@@ -162,6 +162,20 @@ impl Document {
         Ok(doc)
     }
 
+    pub fn reload(&mut self) -> Result<(), anyhow::Error> {
```

Comment:
I think to be safe you might want to implement `Editor::reload` by closing then opening the same file -- if you implement `doc.reload() it needs to reset `history`, `old_state`, `changes`, etc.

Reply:
The `syntax` tree would also need recalculating

Reply:
To keep the same position after `close() & open()` you can copy out `doc.selections` and move them to the new doc (just need to confirm ranges are valid)

Reply:
Oh, right. That makes sense. I'll try that instead. 🙂

Reply:
Hmm, you would have to bypass some of the code in `close`/`open` since it would give you a different id and it would also close the splits. Tricky

Reply:
I've adressed the reset of state in https://github.com/helix-editor/helix/pull/177/commits/293e13ce9417836cd0a33803cb34df8823d70872.